taskcluster/ci/test/kind.yml

```diff
@@ -8,38 +8,56 @@ transforms:
     - taskgraph.transforms.job:transforms
     - taskgraph.transforms.task:transforms
 
+job-defaults:
+    attributes:
+        build-type: debug
+        code-review: true
+    run:
+        # TODO Generate APKs in a build task instead
+        gradlew: ['clean', 'assembleGeckoNightly', 'assembleAndroidTest', 'assembleDebug']
+        secrets:
+            - name: project/mobile/android-components/firebase
+              key: firebaseToken                  
+              path: .firebase_token.json
+              json: true
+        using: gradlew
+        use-caches: false
+    run-on-tasks-for: [github-push]
+    treeherder:
+            kind: test
+            platform: 'ui-test/opt'
+            tier: 2
+    worker-type: b-android
+    worker:
+        artifacts:
+            - name: public
+              path: /build/android-components/results
+              type: directory
+        docker-image: {in-tree: ui-tests}
+        env:
+            GOOGLE_APPLICATION_CREDENTIALS: '.firebase_token.json'
+            GOOGLE_PROJECT: moz-android-components-230120
+        max-run-time: 7200
+
 jobs:
     ui:
```

Comment:
I think my only nit here is that we should rename this job: This is effectively running the tests for `browser-engine-gecko-nightly`. In this case the test APK is important and not the actual app APK. We are only passing an app APK to Firebase because that's required but actually we are just running on-device unit tests for the component. I think we may end up having more of those for our components.